chia/plotting/cache.py

```diff
@@ -0,0 +1,173 @@
+
+        pool_public_key: Optional[G1Element] = None
+        pool_contract_puzzle_hash: Optional[bytes32] = None
+        if isinstance(pool_public_key_or_puzzle_hash, G1Element):
+            pool_public_key = pool_public_key_or_puzzle_hash
+        else:
+            assert isinstance(pool_public_key_or_puzzle_hash, bytes32)
+            pool_contract_puzzle_hash = pool_public_key_or_puzzle_hash
+
+        local_sk = master_sk_to_local_sk(local_master_sk)
+
+        plot_public_key: G1Element = ProofOfSpace.generate_plot_public_key(
+            local_sk.get_g1(), farmer_public_key, pool_contract_puzzle_hash is not None
+        )
+
+        return CacheEntry(
+            prover, farmer_public_key, pool_public_key, pool_contract_puzzle_hash, plot_public_key, time.time()
+        )
+
+    def bump_last_use(self) -> None:
+        self.last_use = time.time()
+
+    def expired(self, expiry_seconds: int) -> bool:
+        return time.time() - self.last_use > expiry_seconds
+
+
+class Cache:
+    _changed: bool
+    _data: Dict[Path, CacheEntry]
+    expiry_seconds: int = 7 * 24 * 60 * 60  # Keep the cache entries alive for 7 days after its last access
+
+    def __init__(self, path: Path) -> None:
+        self._changed = False
+        self._data = {}
+        self._path = path
+        if not path.parent.exists():
+            mkdir(path.parent)
+
+    def __len__(self) -> int:
+        return len(self._data)
+
+    def update(self, path: Path, entry: CacheEntry) -> None:
+        self._data[path] = entry
+        self._changed = True
+
+    def remove(self, cache_keys: List[Path]) -> None:
+        for key in cache_keys:
+            if key in self._data:
+                del self._data[key]
+                self._changed = True
+
+    def save(self) -> None:
+        try:
+            disk_cache_entries: Dict[str, DiskCacheEntry] = {
+                str(path): DiskCacheEntry(
+                    bytes(cache_entry.prover),
+                    cache_entry.farmer_public_key,
+                    cache_entry.pool_public_key,
+                    cache_entry.pool_contract_puzzle_hash,
+                    cache_entry.plot_public_key,
+                    uint64(int(cache_entry.last_use)),
+                )
+                for path, cache_entry in self.items()
+            }
+            disk_cache: DiskCache = DiskCache(
+                uint16(CURRENT_VERSION), [(plot_id, cache_entry) for plot_id, cache_entry in disk_cache_entries.items()]
+            )
+            serialized: bytes = bytes(disk_cache)
+            self._path.write_bytes(serialized)
+            self._changed = False
+            log.info(f"Saved {len(serialized)} bytes of cached data")
+        except Exception as e:
+            log.error(f"Failed to save cache: {e}, {traceback.format_exc()}")
+
+    def load(self) -> None:
+        try:
+            serialized = self._path.read_bytes()
+            version = uint16.from_bytes(serialized[0:2])
```

Comment:
Makes sense, see 7b6a76f16feaf48c6532102135a8f6c87bca3091. It's fine to add it now since this PR anyway was about to break compatibility. Now this makes migrations in the future cleaner so why not :)